modules/ximgproc/src/radon_transform.cpp

```diff
@@ -0,0 +1,421 @@
+ * (including, but not limited to, procurement of substitute goods or services;         *
+ * loss of use, data, or profits; or business interruption) however caused              *
+ * and on any theory of liability, whether in contract, strict liability,               *
+ * or tort (including negligence or otherwise) arising in any way out of                *
+ * the use of this software, even if advised of the possibility of such damage.         *
+ ****************************************************************************************/
+
+ /**
+  * @author {aravind | arvindsuresh2009@gmail.com}
+  * Created on Sat, Feb 13 2016
+  */
+
+ /**
+  * Implementation of "The Radon Transform - Theory and Implementation"
+  *   						 by "Toft, {Peter Aundal} and Sørensen, {John Aasted}"
+  *
+  * Online publication: http://petertoft.dk/PhD/
+  */
+
+// Necessary headers
+
+#include "precomp.hpp"
+
+#include <math.h>
+#include <stdlib.h>
+#include <string>
+#include <iostream>
+
+#include <opencv2/imgproc.hpp>
+#include "opencv2/ximgproc.hpp"
+
+#ifdef __cplusplus
+namespace cv {
+  namespace ximgproc {
+#endif
+
+/**
+ * [trans Function for computing radon transform given a particular angle theta]
+ * @param src        [Source image]
+ * @param dst        [Destination image]
+ * @param m          [Half the columns in src]
+ * @param n          [Half the rows in src]
+ * @param rhoMax     [Maximum value of rho ( restricted due to finite dimension of the image )]
+ * @param theta      [Given value of theta]
+ * @param minAngle   [Minimum values of angle for which transform is computed]
+ */
+
+template <typename T, int D>
+static inline void trans( Mat src,
+                          Mat & dst,
+                          int m,
+                          int n,
+                          int rhoMax,
+                          int theta,
+                          int minAngle
+                        ) {
+  int rc = rhoMax/2;
+
+  // Computing cos(theta) and sin(theta)
+  float cosTh = static_cast<float>(std::cos(theta*CV_PI/180.0));
+  float sinTh = static_cast<float>(std::sin(theta*CV_PI/180.0));
+
+  // Slope of the line = tan(90 - theta)
+  float a = -cosTh/sinTh;
+    // Range: [1, 45]
+    if( theta <= 45 ) {
+      for(int r = 1; r <= rhoMax; ++r) {
+        float rho = (float)(r - rc);
+        float b = rho/sinTh;
+
+        int yMax = (std::min((int)(-a*m + b), n-1)),
+            yMin = (std::max((int)(a*m + b), -n));
+
+        for(int y = yMin; y <= yMax; ++y) {
+          float x = (y - b)/a;
+          int xFloor = static_cast<int>(std::floor(x));
+          float xUp = x - xFloor;
+          float xLow = 1.0f - xUp;
+          x = (float)(xFloor);
+          x = static_cast<float>(std::max((int)(x), -m));
+          x = static_cast<float>(std::min((int)(x), m-2));
+
+          dst.at<float>(rhoMax - r, theta-minAngle) += (float)(xLow)*static_cast<float>(src.at<T>(y + n, (int)(x) + m))
```

Comment:
Since `dst` is used as accumulator here it would be better to fill it with 0 prior to evaluation